### ISSUE TITLE
Implement new transform_num() method: 'normalize'

### ISSUE THREAD

**Issue by ETA444** [ETA444/datasafari#56]
### Description:
---
#### Method Functionality Idea:
The `normalize` transformation method scales numerical variables to a range between 0 and 1. This scaling is beneficial for models sensitive to the scale of variables, enhancing model performance by ensuring all features are on a similar scale.

#### How it operates:
The method utilizes Min-Max Scaling to adjust each numerical variable to the [0, 1] range based on its minimum and maximum values. It returns a DataFrame containing the normalized numerical variables.

#### Usage:
To normalize numerical variables to the [0, 1] range:
```python
transformed_df, normalized_columns = transform_num(your_df, your_numerical_variables, method='normalize')
```
This method returns both the DataFrame with normalized numerical variables and a DataFrame containing only the normalized columns.

#### Notes:
- It's important to ensure the data is clean and outliers are handled appropriately before applying normalization for optimal results.
- Prior data inspection and outlier detection using methods like `explore_num()` are recommended.

**Comment by ETA444** [comment]
**Implementation Summary:**

The `'normalize'` method scales numerical variables to a [0, 1] range, making them suitable for models sensitive to variable scales.

**Code Breakdown:**

1. **Method Header:**

   - **Purpose:** Clearly indicate the start of the `'normalize'` method implementation and provide context.

   ```python
   if method.lower() == 'normalize':
       print(f"< NORMALIZATION TRANSFORMATION >")
       print(f" This method scales numerical variables to a [0, 1] range, making them suitable for models sensitive to variable scales.")
       print(f"  ✔ Adjusts each feature to a [0, 1] scale based on its minimum and maximum values.")
       print(f"  ✔ Enhances model performance by ensuring numerical variables are on a similar scale.")
       print(f"✎ Note: Ensure data is clean and outliers are handled for optimal results.\n")
       print(f"☻ Tip: Use `explore_num()` for data inspection and outlier detection before applying normalization.\n")
   ```

2. **Initialize Min-Max Scaler:**

   - **Purpose:** Prepare the necessary data structures for transformation.

   ```python
   # initialize essentials
   transformed_df = df.copy()
   min_max_scaler = MinMaxScaler()
   ```

3. **Apply Min-Max Scaling:**

   - **Purpose:** Apply the transformation to scale the specified numerical variables.

   ```python
   # apply minmax scaling
   transformed_df[numerical_variables] = min_max_scaler.fit_transform(df[numerical_variables])
   ```

4. **Output Results:**

   - **Purpose:** Display the results and perform a sanity check.

   ```python
   # isolate transformed columns to provide as part of output
   normalized_columns = transformed_df[numerical_variables]

   print(f"✔ New transformed dataframe:\n{transformed_df.head()}\n")
   print(f"✔ Dataframe with only the transformed columns:\n{normalized_columns.head()}\n")
   print("☻ HOW TO: Apply this transformation using `transformed_df, normalized_columns = transform_num(your_df, your_numerical_variables, method='normalize')`.\n")

   # sanity check
   print("< SANITY CHECK >")
   print(f"  ➡ Shape of original dataframe: {df.shape}")
   print(f"  ➡ Shape of transformed dataframe: {transformed_df.shape}\n")

   return transformed_df, normalized_columns
   ```

**Link to Full Code:** [transform_num.py](https://github.com/ETA444/datasafari/blob/main/datasafari/transformer/transform_num.py)
